addons/mail/static/src/core/common/composer.js

```diff
@@ -539,9 +539,20 @@ export class Composer extends Component {
             // Reset signature when recovering an empty body.
             composer.emailAddSignature = true;
         }
+        let signature = this.store.self.signature;
+        if (signature) {
+            const parser = new DOMParser();
+            const doc = parser.parseFromString(signature, 'text/html');
+            const spanElement = document.createElement('span');
+            spanElement.setAttribute('data-o-mail-quote', '1');
+            const br = document.createElement("br");
+            const textNode = document.createTextNode("-- ");
+            spanElement.append(textNode, br, ...doc.body.childNodes);
```

Comment:
Hi @spe-odoo I don't think it's ok to put the content of a HTML field inside a `span` node:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/span#technical_summary

It may generate malformed HTML as `<span>` element permitted content is restrictive to phrasing content (https://developer.mozilla.org/en-US/docs/Web/HTML/Content_categories#phrasing_content).

Why not put everything in a `<div>` instead ?

Reply:
Will do it while merging.